crates/rust-analyzer/src/main_loop.rs

```diff
@@ -389,19 +399,32 @@ fn loop_turn(
                 req,
             )?,
             Message::Notification(not) => {
-                on_notification(
-                    &connection.sender,
-                    world_state,
-                    &mut loop_state.pending_requests,
-                    &mut loop_state.subscriptions,
-                    not,
-                )?;
+                on_notification(&connection.sender, world_state, loop_state, not)?;
             }
             Message::Response(resp) => {
                 let removed = loop_state.pending_responses.remove(&resp.id);
                 if !removed {
                     log::error!("unexpected response: {:?}", resp)
                 }
+
+                if Some(resp.id) == loop_state.configuration_request_id {
+                    loop_state.configuration_request_id = None;
+                    if let Some(err) = resp.error {
+                        log::error!("failed to fetch the server settings: {:?}", err)
+                    } else if let Some(result) = resp.result {
+                        let new_config = serde_json::from_value::<Vec<ServerConfig>>(result)?
```

Comment:
Does this have to be a `Vec`? I don't see that the lsp protocol says it has to be so, its just `any`, not `any[]`

Reply:
> it's just any, not any[]

It's `any[]`: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspace_configuration

```
Response:

    result: any[]
    error: code and message set in case an exception happens during the ‘workspace/configuration’ request
```

Reply:
Ah, yeah, looking at the wrong thing...